Frontend/ui-customer/src/Flow.purs

```diff
@@ -2051,41 +2067,70 @@ homeScreenFlow = do
 
-        gateAddress =
-          if DS.null state.props.defaultPickUpPoint then
-            HomeScreenData.dummyLocation
-          else
-            fromMaybe HomeScreenData.dummyLocation (Arr.find (\pickupPoint -> pickupPoint.place == state.props.defaultPickUpPoint) pickUpPoints)
-      setValueToLocalStore CUSTOMER_LOCATION $ show cityName
-      checkForSpecialZoneAndHotSpots state (ServiceabilityRes sourceServiceabilityResp) lat lon
+        gateAddress = case isSource of 
+                            Source -> if DS.null state.props.defaultPickUpPoint then
+                                HomeScreenData.dummyLocation
+                              else
+                                fromMaybe HomeScreenData.dummyLocation (Arr.find (\pickupPoint -> pickupPoint.place == state.props.defaultPickUpPoint) pickUpOrDropOfPoints)
+                            Destination -> 
+                              if DS.null state.props.defaultDropOfPoint then
+                                  HomeScreenData.dummyLocation
+                              else
+                                fromMaybe HomeScreenData.dummyLocation (Arr.find (\pickupPoint -> pickupPoint.place == state.props.defaultDropOfPoint) pickUpOrDropOfPoints)
+      case isSource of 
+        Source -> setValueToLocalStore CUSTOMER_LOCATION (show cityName)
+        Destination -> pure unit
+
+      checkForSpecialZoneAndHotSpots state (ServiceabilityRes locServiceabilityResp) lat lon isSource
       let
-        distanceBetweenLatLong = getDistanceBwCordinates lat lon state.props.locateOnMapLocation.sourceLat state.props.locateOnMapLocation.sourceLng
+        cachedLat = (case isSource of
+                        Source -> state.props.locateOnMapLocation.sourceLat
+                        Destination -> state.props.locateOnMapLocation.destinationLat)
+        cachedLon = (case isSource of
+                        Source -> state.props.locateOnMapLocation.sourceLng
+                        Destination -> state.props.locateOnMapLocation.destinationLng)
+        distanceBetweenLatLong = getDistanceBwCordinates lat lon cachedLat cachedLon
 
         isMoreThan20Meters = distanceBetweenLatLong > (state.data.config.mapConfig.locateOnMapConfig.apiTriggerRadius / 1000.0)
       modifyScreenState
-        $ HomeScreenStateType
-            ( \homeScreen ->
-                homeScreen
+        $ HomeScreenStateType ( 
+                  \homeScreen -> homeScreen 
                   { props
-                    { sourceLat = lat
-                    , sourceLong = lon
-                    , editedPickUpLocation {gps = LatLong {lat : lat
-                                                    , lon : lon 
-                                                    }  
-                                  }
-                    , confirmLocationCategory = getZoneType srcSpecialLocation.category
-                    , city = cityName
+                    { sourceLat = case isSource of
```

Comment:
can we create a props object here for `Source` and `Destination` instead of multiple case matches?